### PR TITLE
[style] frontend age pref question 👵👶 

### LINKED ISSUE #140: style frontend for age preference question
yay great job last week! 🍊

This week, you’ll be styling the age offense question [Jesslyn added in hi-fi’s here](https://www.figma.com/design/fPzdPfT77bx4wXHBkAQbj6/Adopt-an-Inmate?node-id=1-3&p=f&t=ZCkfVwoGuYgHYhwW-0). 

Your work will build off of [Sam’s previous sprint](https://github.com/calblueprint/adopt-an-inmate/issues/132) working on the backend for this question. Most of the code for linking this question to the backend should already be complete for you, but if anything is confusing, follow the precedent set by our other main application questions (like `MainQuestionBio.tsx` and `MainQuestionGender.tsx`).

### PR DESCRIPTION
**Pull request by plaanty**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Styled the age preference question page within the main app according to Jesslyn's figma designs and changed the backend to match.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

For "NO" age pref
Screenshot 2026-03-18 073045

For "YES" age pref, but without moving sliders
Screenshot 2026-03-18 073102

For "YES" age pref, with moving sliders
Screenshot 2026-03-18 073117

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Make a test application, walk through it to make sure the UI is responsive and is sufficiently resembling the figma file.
Check that the review page contains the age preference information in the way that we want.
Check that the test application data is recorded accurately in the supabase.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Will potentially need to rebase when merging (rip)
- Need to slightly change the style of the default max age (Jinkang told me how to do it kinda, it just takes a very convoluted process of laying an invisible element on top of a visible one, etc)
- Right now, by default the page is set to "NO" as the response, might need to fix this

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @InayaY3

### CODE REVIEW

**Review by jinkang-0 — COMMENTED** [review]
looks great so far, just have a few nits and concerns

File: components/application/main/MainQuestionReview.tsx [hunk 1]
```diff
@@ -91,8 +92,17 @@ export default function MainQuestionReview() {
...
+            <p className="text-[0.78844rem] leading-normal font-[600] font-medium text-[var(--color-gray-8)]">
+              Age range preference
+            </p>
+            <p className="text-[1.0035rem] leading-normal font-[400] font-medium text-[var(--color-gray-12)]">
+              {formatAgePreference(appState.form.agePreference)}
+            </p>
+          </div>
+
+          <div className="flex flex-col gap-[0.38]">
+            <p className="text-[0.78844rem] leading-normal font-[600] font-medium text-[var(--color-gray-8)]">
```
Comment (InayaY3) [inline components/application/main/MainQuestionReview.tsx]: can you fix the styling here to match that of the rest of the page? it currently shows up as bold (and sai has since fixed the tailwind classes). also fix the formatting of the age pref as mentioned

File: components/application/main/MainQuestionAge.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+              </div>
+
+              <div className="flex flex-col gap-1">
+                <p className="text-right text-gray-10">Maximum</p>
+                <input
+                  type="number"
+                  className="h-9 rounded-lg border-2 border-gray-10"
+                  value={maxAge ?? 80}
+                  onChange={e => setValue('maxAge', parseInt(e.target.value))}
+                />
```
Comment (InayaY3) [inline components/application/main/MainQuestionAge.tsx]: - i'm able to change the input for the text boxes to be less than 18 and greater than 80 (not just through manual input, even with the arrows). you should be setting props here for `min` and `max` to prevent this at least through the arrows. 
- would also set `maxLength` to prevent numbers more than double digits. both boxes prevent letters/chars which is good. 
- also you can use our existing `Textbox` component instead of input, i believe the styling should match. 
- i get an error whenever the boxes are cleared “Received NaN for the `value` attribute. If this is expected, cast the value to a string.” since you are parsing within the inputs. you should handle empty boxes or reset them to defaults.
- `OnboardingQuestionAdoptedBefore.tsx` has similar inputs as an example (for some of the fts, not all)

File: components/application/main/MainQuestionAge.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  const { appState, setAppState } = useApplicationContext();
+  const { nextQuestion } = useQuestionNavigaton();
+  const { upsertAppInfo } = useApplicationNavigation();
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setValue,
+    //formState: { errors },
```
Comment (InayaY3) [inline components/application/main/MainQuestionAge.tsx]: add in an `ErrorMessage` the way other application questions have. you can set the error messages for `minAge` and `maxAge` within `agePrefFormSchema` above

**Review by InayaY3 — CHANGES_REQUESTED** [review]
the overall functionality looks good so far! 

i've requested some changes for styling, maintaining `appState` for question navigation, types, etc. on the frontend. the backend seems to be working okay aside from the `string[]` vs `int[]` issue and not setting the col val to null for no age pref.

i've rebased for you and fixed the lint warnings, but the rest of the changes are too substantial for me to quickly fix for you. defo follow the comments, check out onboarding/other main app questions for partial examples, and lmk if you need help! 

*also note: i've closed [your backend PR for age pref](https://github.com/calblueprint/adopt-an-inmate/pull/142) so just focus on making the requested changes here! this PR has full functionality for the age pref question (aside from altering the find top k function on supabase, which i will prob do). check out my comments on that PR too tho if you haven't already. they may apply here

File: components/application/main/MainQuestionAge.tsx [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+  const minAge = watch('minAge');
+  const maxAge = watch('maxAge');
+
+  const onSubmit = async ({
+    hasAgePreference,
+    minAge,
+    maxAge,
+  }: z.infer<typeof agePrefFormSchema>) => {
+    console.log('typeof minAge:', typeof minAge, 'value:', minAge);
+    console.log('typeof maxAge:', typeof maxAge, 'value:', maxAge);
```
Comment (jinkang-0) [inline components/application/main/MainQuestionAge.tsx]: should remove console logs before deployment

File: actions/queries/query.ts [hunk 5]
```diff
@@ -84,6 +84,10 @@ export async function upsertApplication(
   if (error)
     throw new Error(`Error upserting application data: ${error.message}`);
 
+  console.log('upsertAppInfo received:', JSON.stringify(app));
+  console.log('userId:', app.adopter_uuid);
+  console.log('appId:', app.app_uuid);
```
Comment (jinkang-0) [inline actions/queries/query.ts]: should remove console logs

File: components/application/main/MainQuestionAge.tsx [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+            >
+              <Slider.Track className="relative h-2 w-full grow overflow-hidden rounded-full bg-gray-200">
+                <Slider.Range className="absolute h-full bg-[#B73940]" />
+              </Slider.Track>
+
+              <Slider.Thumb className="block h-6 w-6 rounded-full border border-red-900 bg-white shadow" />
+              <Slider.Thumb className="block h-6 w-6 rounded-full border border-red-900 bg-white shadow" />
+            </Slider.Root>
+
+            <div className="flex flex-row justify-between">
```
Comment (jinkang-0) [inline components/application/main/MainQuestionAge.tsx]: should establish minimum gap between the two textboxes

otherwise, small screens will show:

Image

File: components/application/main/MainQuestionAge.tsx [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+              <div className="flex flex-col gap-1">
+                <p className="text-gray-10">Minimum</p>
+                <Textbox
+                  className="h-9 rounded-lg border-2 border-gray-10 pl-2"
```
Comment (jinkang-0) [inline components/application/main/MainQuestionAge.tsx]: should be `border-1` to match Figma

File: components/application/main/MainQuestionAge.tsx [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+                    <p className="text-sm text-red-600">
+                      {errors.minAge.message}
+                    </p>
+                  )}
+              </div>
+
+              <div className="flex flex-col gap-1">
+                <p className="text-right text-gray-10">Maximum</p>
+                <Textbox
+                  className="h-9 rounded-lg border-2 border-gray-10 pr-2 text-right"
```
Comment (jinkang-0) [inline components/application/main/MainQuestionAge.tsx]: should be `border-1` to match Figma

File: components/application/main/MainQuestionAge.tsx [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+                setMinAgeDisplay(String(vals[0]));
+                setMaxAgeDisplay(vals[1] === 80 ? '80+' : String(vals[1]));
+              }}
+            >
+              <Slider.Track className="relative h-2 w-full grow overflow-hidden rounded-full bg-gray-200">
+                <Slider.Range className="absolute h-full bg-[#B73940]" />
+              </Slider.Track>
+
+              <Slider.Thumb className="block h-6 w-6 rounded-full border border-red-900 bg-white shadow" />
+              <Slider.Thumb className="block h-6 w-6 rounded-full border border-red-900 bg-white shadow" />
```
Comment (jinkang-0) [inline components/application/main/MainQuestionAge.tsx]: shadow should be more pronounced (ref [tailwind shadow guide](https://tailwindcss.com/docs/box-shadow))

site:

Image

figma:

Image

File: components/application/main/MainQuestionAge.tsx [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+                <p className="text-right text-gray-10">Maximum</p>
+                <Textbox
+                  className="h-9 rounded-lg border-2 border-gray-10 pr-2 text-right"
+                  value={maxAgeDisplay}
+                  maxLength={2}
+                  onChange={e => {
+                    const raw = e.target.value.replace('+', '');
+                    setMaxAgeDisplay(
+                      raw === '' ? '' : parseInt(raw) === 80 ? '80+' : raw,
+                    );
```
Comment (jinkang-0) [inline components/application/main/MainQuestionAge.tsx:213]: would recommend changing the inequality to `parseInt(raw) >= 80` so inputs greater than 80 will also be parsed as "80+"

**Review by jinkang-0 — COMMENTED** [review]
good work sam! added comments for design matching and technicalities. Also unresolved two past comments that weren't resolved.

File: components/Button.tsx [hunk 11]
```diff
@@ -13,6 +13,7 @@ const buttonStyle = cva(
         secondary: 'border border-gray-8 bg-gray-1 rounded-lg text-cyan-12',
         primary: 'bg-cyan-12 rounded-lg text-gray-1',
         tertiary: 'bg-cyan-9 text-white rounded-lg py-3 px-4',
+        quaternary: 'bg-red-9 rounded-lg text-red-1',
```
[the remark this answers is not in the document]
Comment (plaanty) [reply components/Button.tsx:16]: If I update this to be the primary, then it influences a bunch of other button interactions. I think that should be a mini-sprint in and of itself to change this.

**Review by InayaY3 — APPROVED** [review]
lgtm! few notes:
- will need to test the filter being applied more thoroughly once the find top k update PR is merged in
- minor bug (or part of the UX?), but we aren't saving user responses when we click "back" before hitting "next" since we only upsert + update `appState` upon clicking next. not sure abt the best fix. useEffect is kinda bandaid, and idk what the best way to call `upsertApplication` is without refactoring `app-process`. will be a future minor sprint prob!
- my commit just fixed the radix-slider import and any unnecessary changes to our packages/dependency files
